lib/msf/core/post/linux/system.rb

```diff
@@ -270,6 +300,110 @@ def get_mount_path(filepath)
   end
 
 
+  #
+  # Gets all the IP directions of the device
+  # @return [Array]
+  #
+  def ips()
+    lines = read_file("/proc/net/fib_trie")
+    result = []
+    previous_line = ""
+    lines.each_line do |line|
+      if line.include?("/32 host LOCAL")
+        previous_line = previous_line.split("-- ")[1].strip()
+        if not result.include? previous_line
+          result.insert(-1, previous_line)
+        end
+      end
+      previous_line = line
+    end
+    return result
+  end
+
+
+  #
+  # Gets all the interfaces of the device
+  # @return [Array]
+  #
+  def interfaces()
+    result = []
+    data = cmd_exec("for fn in /sys/class/net/*; do echo $fn; done")
+    parts = data.split("\n")
+    parts.each do |line|
+      line = line.split("/")[-1]
+      result.insert(-1,line)
+    end
+    return result
+  end
+
+
+
+  #
+  # Gets all the macs of the device
+  # @return [Array]
+  #
+  def macs()
+    result = []
+    str_macs = cmd_exec("for fn in /sys/class/net/*; do echo $fn; done")
+    parts = str_macs.split("\n")
+    parts.each do |line|
+      rut = line + "/address"
+      mac_array = read_file(rut)
+      mac_array.each_line do |mac|
+        result.insert(-1,mac.strip())
+      end
+    end
+    return result
+  end
+
+
+
+  # Parsing information based on: https://github.com/sensu-plugins/sensu-plugins-network-checks/blob/master/bin/check-netstat-tcp.rb
+  #
+  # Gets all the listening tcp ports in the device
+  # @return [Array]
+  #
+  def listen_tcp_ports()
```

Comment:
Style. No need to use brackets `()` in method definitions when the method takes no arguments.